sdk/core/azure-core/azure/core/pipeline_client.py

```diff
@@ -83,16 +77,14 @@ class PipelineClient(PipelineClientBase):
             :caption: Builds the pipeline client.
     """
 
-    def __init__(self, base_url, config, **kwargs):
+    def __init__(self, base_url, **kwargs):
         super(PipelineClient, self).__init__(base_url)
-        if config is None:
-            raise ValueError("Config is a required parameter")
-        self._config = config
+        self._config = kwargs.get("config", None) or Configuration(**kwargs)
         self._base_url = base_url
         if kwargs.get("pipeline"):
             self._pipeline = kwargs["pipeline"]
         else:
-            self._pipeline = self._build_pipeline(config, **kwargs)
+            self._pipeline = self._build_pipeline(self._config, **kwargs)
```

Comment:
`self._config` does not appear to ever be used except to pass to `_build_pipeline`. Is there actually a reason it needs to to be stored on `self` at all?

Reply:
It may be used in operations.

Reply:
Well, I was asking because it's definitely not used anywhere else in `azure-core`
```
base ❯ grip "\._config"
./azure-core/azure/core/pipeline_client.py:
   90 :         self._config = config
./azure-core/azure/core/pipeline_client_async.py:
   89 :         self._config = config
./azure-core/azure/core/settings.py:
  384 :         return self._config(props)
  405 :         return self._config(props)
```
Do you mean things outside `aure-core` might be accessing this private attribute?

Reply:
e.g. in operations(in generated code), we have code like
if self._config.generate_client_request_id:
                header_parameters['x-ms-client-request-id'] = str(uuid.uuid1())

Reply:
ah, got it, thanks